### ISSUE TITLE
Magnifying glass

### ISSUE THREAD

**Issue by tombh** [tombh/texttop#33]
I'm thinking of a mode where clicking creates a slightly zoomed box of the page under the browser.

**Comment by tobimensch** [comment]
Very low priority and not really necessary in my opinion.

Is the intention of this to zoom into images?

I think it would be more useful if the user had an easy way to get an image's URL. If he really wanted to view an image, he could also load it in a traditional browser

**Comment by tombh** [comment]
I rarely have a need for this, so I agree it's low priority.

I think it'd be good if the magnifier also showed the original text as rendered by the browser, so you could quickly discern any discrepancies. And also it'd be good if mouse clicks where magnified too, so you could click on fiddly little UI buttons.

**Comment by daniellandau** [comment]
My use case for this would be to render QR codes in a way that they are scannable. The main website I'd like to do this on is web.whatsapp.com, but I imagine it'd be useful elsewhere too.

**Comment by tombh** [comment]
Yes! I have exactly the same issue! Imagine being able to use Whatsapp in the terminal :D

**Comment by Yash-Singh1** [comment]
Never mind, I forgot that `Browsh` uses `Firefox`. The `StackOverFlow` answer only is for `Chrome`. To get it working on `Firefox` we need to apply the following CSS:
```css
body {
	-moz-transform: scale(0.4);
	-moz-transform-origin: 0 0
}
```
The `scale` value on `-moz-transform` will have to be changed. Open up `about:blank` and type in the following in the console:
```js
document.body.style['-moz-transform']=<scale>
```
Replace `<scale>` with your scale value. This works perfectly with `Firefox`. Whenever `Ctrl + +` is triggered then we need to send a zoom-in call to the `WebExt` which will run:
```js
document.body.style['-moz-transform']+=0.1
```
Whenever `Ctrl + -` is triggered then we need to send a zoom-in call to the `WebExt` which will run:
```js
document.body.style['-moz-transform']-=0.1
```
At the initialization of a tab, we need to run:
```js
document.body.style['-moz-transform']=1.0
```
On that tab.

**Comment by Yash-Singh1** [comment]
Hey there, I recently added in this feature inside a new project I created: `zoom.js`. Here is the [link](https://github.com/Yash-Singh1/zoom.js) if you want to look at it. If you would like to preview the zooming inside `firefox`, go to the [test link](https://yash-singh1.github.io/zoom.js/test/interactive.html) and run:
```javascript
test()
```
Inside the console.
Also, the previous comment I put had some mistakes. We would need to run:
```javascript
document.body.style['-moz-transform-origin'] = "0 0";
document.body.style['-moz-transform'] = "scale(1)";
```
At the initialization of each tab. Every time we zoom-in:
```javascript
let newFloat = parseFloat(document.body.style["-moz-transform"].substring(6,document.body.style["-moz-transform"].length - 1))
newFloat += 1
document.body.style["-moz-transform"] = "scale(" + newFloat + ")";
```
Everytime we zoom-out:
```javascript
let newFloat = parseFloat(document.body.style["-moz-transform"].substring(6,document.body.style["-moz-transform"].length - 1))
newFloat -= 1
document.body.style["-moz-transform"] = "scale(" + newFloat + ")";
```

**Comment by niutech** [comment]
You no longer need to use `-moz-transform`, Firefox supports `transform`.

**Comment by Mart-Bogdan** [comment]
I'm just thinking that we could add a hotkey to force FF to render visible screen into jpg/png/pdf into the current directory, which could be fetched by ssh(sftp/scp) and viewed on the client machine. would be useful for captchas.  And other use-cases are possible too,

**Comment by Aster89** [comment]
> Yes! I have exactly the same issue! Imagine being able to use Whatsapp in the terminal :D

I don't remember exactly what was like in 2018, but jeez, today we use QR code to log into so many things!
